### PR TITLE
my log stuff

### PR DESCRIPTION
**Pull request by Bluscream**
I splitted LogCmd into LogCmd.In and LogCmd.Out and also added LogCmd.Server for my case as 4netplayers doesn't provide proper logging

### CODE REVIEW

File: TS3AudioBot/NLog.config [hunk 1]
```diff
@@ -3,25 +3,34 @@
...
       throwConfigExceptions="true">
 
-	<targets>
-		<!-- Config log to file -->
-		<target layout="${longdate}|${pad:padding=5:inner=${level:uppercase=true}}|${callsite:includeNamespace=false} ${message}${onexception:${newline}${exception:format=tostring}}" name="logfile" xsi:type="File" fileName="ts3audiobot.log" />
-		<!-- Config log to console -->
-		<target xsi:type="ColoredConsole"
+  <targets>
+    <!-- Config log to file -->
+    <target layout="${longdate}|${pad:padding=5:inner=${level:uppercase=true}}|${callsite:includeNamespace=false} ${message}${onexception:${newline}${exception:format=tostring}}" name="logfile" xsi:type="File" fileName="/var/log/ts3ab/bot.log" />
```
Comment (Splamy) [inline TS3AudioBot/NLog.config]: Please undo this, not everyone runs this on a linux server.

File: TS3AudioBot/NLog.config [hunk 2]
```diff
@@ -3,25 +3,34 @@
...
-		<logger name="TS3AudioBot.*" minlevel="Info" writeTo="logfile" />
-		<logger name="TS3AudioBot.*" minlevel="Info" writeTo="console" />
-		<logger name="TS3Client.*" minlevel="Info" writeTo="logfile" />
-		<logger name="TS3Client.*" minlevel="Info" writeTo="console" />
-	</rules>
+  <rules>
+    <logger name="TS3AudioBot.Plugins.*" minlevel="Trace" writeTo="logfile" />
+    <logger name="TS3AudioBot.*" minlevel="Debug" writeTo="logfile" />
+    <logger name="TS3AudioBot.*" minlevel="Info" writeTo="console" />
+    <logger name="TS3Client" minlevel="Info" writeTo="logfile" />
```
Comment (Splamy) [inline TS3AudioBot/NLog.config]: There is no loggaer with the name TS3Client afaik, that is the namespace

File: TS3AudioBot/NLog.config [hunk 3]
```diff
@@ -3,25 +3,34 @@
...
+    <logger name="TS3Client.PacketHandler.Rtt" minlevel="Warn" writeTo="logfile" />
+    <logger name="TS3Client.PacketHandler.Raw" minlevel="Warn" writeTo="logfile" />
+    <logger name="TS3Client.PacketHandler.Raw.Voice" minlevel="Warn" writeTo="logfile" />
+    <logger name="TS3Client.PacketHandler.Timeout" minlevel="Warn" writeTo="logfile" />
```
Comment (Splamy) [inline TS3AudioBot/NLog.config]: Any explicit reason why you push all these loggers default to `Warn`?

File: TS3Client/Full/Ts3FullClient.cs [hunk 4]
```diff
@@ -186,7 +187,17 @@ private void PacketEvent(ConnectionContext ctx, ref Packet<S2C> packet)
...
+					LogCmdIn.Trace("[I] {0}", str);
+					if (str.StartsWith("notify")) {
+							if (str != lastnotify)
+							{
+								lastnotify = str;
+								LogCmdServer.Debug(str);
+							} else {
+								LogCmdServer.Debug("+1");
+							}
+						}
```
Comment (Splamy) [inline TS3Client/Full/Ts3FullClient.cs]: This block is a horrible hack; doesn't catch all notifications, produces allocations, and overhead. And doesn't even save that much readability as in a normal console where you can change the `+1` to a `+2`

File: TS3Client/Generated/Messages.cs [hunk 5]
```diff
@@ -9467,6 +9467,8 @@ public enum NotificationType
 		VersionRequest,
 		///<summary>[C2S] ntfy:whoami</summary>
 		WhoAmIRequest,
+		///<summary>[S2C] ntfy:any</summary>
+		RawCommand,
```
Comment (Splamy) [inline TS3Client/Generated/Messages.cs]: You do realize that's not how it works. Those are autogenerated functions. And just adding a `RawCommand` will not work. If you that desperately need a raw command, I can see what I can do there.

File: TS3AudioBot/NLog.config (hunk 1 above)
Comment (Bluscream) [reply TS3AudioBot/NLog.config]: my intention was not for you to push this but for me to have a reference on how i did it even when i delete my fork ^^

File: TS3AudioBot/NLog.config (hunk 3 above)
Comment (Bluscream) [reply TS3AudioBot/NLog.config]: Because i only want their warnings.

File: TS3Client/Full/Ts3FullClient.cs (hunk 4 above)
Comment (Bluscream) [reply TS3Client/Full/Ts3FullClient.cs]: It kinda works for me, i would prefer to have it logged through a plugin to a Sql(ite) database though (https://github.com/Bluscream/TS3AudioBotPlugins/blob/da3dacfe4363d147b88f7e7249295e692ea91632/ServerLog/Plugin.cs). My attempt with litedb wasn't really successfull either (https://github.com/Bluscream/TS3AudioBotPlugins/blob/d1cd1442e8792b948b3d15a1f9e95083d206a73c/ServerLog/Plugin.cs).

File: TS3Client/Generated/Messages.cs (hunk 5 above)
Comment (Bluscream) [reply TS3Client/Generated/Messages.cs]: Yes, please. for simplicity sake

File: TS3AudioBot/NLog.config (hunk 3 above)
Comment (Splamy) [reply TS3AudioBot/NLog.config]: Then don't make a pull request with those changes?

**Pull request closed without merging**